spec-tests/src/test/java/org/candlepin/spec/entitlements/StorageBandSpecTest.java

```diff
@@ -0,0 +1,256 @@
+            .isNotNull()
+            .returns("partial", ComplianceStatusDTO::getStatus)
+            .returns(false, ComplianceStatusDTO::getCompliant);
+    }
+
+    @Test
+    public void shouldStorageBandEntitlementsFromSameSubscriptionCanBeStackedToCoverEntireSystem() {
+        OwnerDTO owner = ownerApi.createOwner(Owners.random());
+        ApiClient userClient = ApiClients.basic(UserUtil.createUser(client, owner));
+        ProductDTO cephProduct = createCephProductAndPool(owner);
+
+        ConsumerDTO system = userClient.consumers().createConsumer(
+            Consumers.random(owner)
+            .type(ConsumerTypes.System.value())
+            .facts(Map.of("band.storage.usage", "256"))
+            .installedProducts(Set.of(new ConsumerInstalledProductDTO()
+            .productId(cephProduct.getId()).productName(cephProduct.getName()))));
+        ApiClient systemClient = ApiClients.ssl(system);
+
+        PoolDTO cephPool = userClient.pools().listPoolsByOwnerAndProduct(owner.getId(),
+            cephProduct.getId()).get(0);
+
+        // Partial stack
+        systemClient.consumers().bindPool(system.getUuid(), cephPool.getId(), 128);
+        assertThat(systemClient.consumers().getComplianceStatus(system.getUuid(), null))
+            .isNotNull()
+            .returns("partial", ComplianceStatusDTO::getStatus)
+            .returns(false, ComplianceStatusDTO::getCompliant);
+
+        // Complete the stack
+        systemClient.consumers().bindPool(system.getUuid(), cephPool.getId(), 128);
+        assertThat(systemClient.consumers().getComplianceStatus(system.getUuid(), null))
+            .isNotNull()
+            .returns("valid", ComplianceStatusDTO::getStatus)
+            .returns(true, ComplianceStatusDTO::getCompliant);
+
+        List<EntitlementDTO> entitilements = userClient.consumers().listEntitlements(system.getUuid());
+        assertThat(entitilements)
+            .hasSize(2)
+            .filteredOn(x -> x.getQuantity() == 128)
+            .hasSize(2);
+    }
+
+    @Test
+    public void shouldStorageBandEntitlementsAutoAttachCorrectQuantity() {
+        OwnerDTO owner = ownerApi.createOwner(Owners.random());
+        ApiClient userClient = ApiClients.basic(UserUtil.createUser(client, owner));
+        ProductDTO cephProduct = createCephProductAndPool(owner);
+
+        ConsumerDTO system = userClient.consumers().createConsumer(
+            Consumers.random(owner)
+            .type(ConsumerTypes.System.value())
+            .facts(Map.of("band.storage.usage", "256"))
+            .installedProducts(Set.of(new ConsumerInstalledProductDTO()
+            .productId(cephProduct.getId()).productName(cephProduct.getName()))));
+        ApiClient systemClient = ApiClients.ssl(system);
+
+        assertThat(systemClient.consumers().bindProduct(system.getUuid(), cephProduct.getId()))
+            .hasSize(1)
+            .singleElement()
```

Comment:
`.singleElement()` already asserts that the size is 1, so we should be able to remove `hasSize(1)`